doc/dart2_migration.md

```diff
@@ -0,0 +1,279 @@
+
+This is solved by providing an alternative API to obtain the same information.
+Because the builder is generating the public, un-prefixed versions of all the
+Props and State classes and mixins, it can include a static `meta` field:
+
+```dart
+// generated by builder
+class FooProps extends _$FooProps with _$FooPropsAccessorsMixin {
+  static PropsMeta meta = $metaForFooProps;
+}
+
+const PropsMeta $metaForFooProps = const PropsMeta(
+  fields: ...,
+  keys: ...,
+);
+```
+
+With this in place, existing usages of `$Props()` and `$PropKeys()` should be
+migrated like so:
+
+```diff
+- const $Props(FooProps)
++ FooProps.meta
+
+- const $PropKeys(FooProps)
++ FooProps.meta.keys
+```
+
+As an added bonus, this meta information becomes available for State classes as
+well.
+
+### Component Default Props
+
+In some cases, obtaining the default props for a component can be useful. These
+are defined via the component's `getDefaultProps()` method. The easiest way to
+obtain these default props is to simply construct the component and call that
+method. Unfortunately, constructing the component directly was never intended to
+be a supported use case, and in Dart 2 this approach won't work because the
+typed props factory implementation needs to be generated.
+
+To work around around this, an alternative API for obtaining a component's
+default props has been added:
+
+```diff
+- var defaultProps = new FooComponent().getDefaultProps();
++ var defaultProps = Foo().componentDefaultProps;
+```
+
+The `componentDefaultProps` getter returns the cached default props for the
+component that the factory would eventually construct. One caveat: the getter is
+typed as `Map`. If you need to obtain a component's default props typed as the
+props class for that component, use the following utility:
+
+```diff
+- var defaultProps = new FooComponent().getDefaultProps();
++ var defaultProps = typedDefaultPropsFor(Foo);
+```
+
+## Migration From Dart 1 to Dart 2
+
+Now that we've laid out the eventual destination, we need to explain how to get
+there.
+
+If you don't need to support a backwards-compatible migration path and just want
+to get from Dart 1 to Dart 2 as quickly as possible, you have two options:
+
+1. Update your code manually using the above diffs as a guide.
+2. Use our [`over_react_migrate_from_dart1_to_dart2` codemod script][orcm-dart1-to-dart2]
+   to automate the migration.
+
+If, however, you do need migrate your `over_react` code from Dart 1 to Dart 2
+in a backwards- and forwards-compatible manner, you'll need to take a two-step
+approach:
+
+### 1. Migrate to the Forwards- and Backwards-compatible Setup
+
+Use our [`over_react_migrate_to_dart1_and_dart2` codemod script][orcm-dart1-to-dart1-and-dart2]
+to update your code to a state that is compatible with both the Dart 1
+transformer and the Dart 2 builder. _In this state, you will notice some extra
+boilerplate and comments. This will be cleaned up in the second step, but is
```

Comment:
```suggestion
boilerplate and comments. This will be cleaned up when the transition to dart 2 is completed and we can pull the plug on dart 1 compatibility, but is
```